File: 20260521/exos/02_condition.py
# ============================================================
# Resident Evil — porte verrouillée
# ============================================================
# Leon arrive devant une porte. Si le niveau d'accès est au moins 3,
# la porte s'ouvre. Sinon elle reste fermée.
#
# Lance : python3 02_condition.py
# ============================================================

niveau_acces = 3
etat_porte = ""

# Détermine etat_porte selon niveau_acces.
#
# niveau_acces = 3  →  3 >= 3 ? oui  →  etat_porte = "Ouverte"
# niveau_acces = 1  →  1 >= 3 ? non  →  etat_porte = "Fermée"
#
# Résultat attendu : etat_porte == "Ouverte"  (car niveau_acces vaut 3 ici)
#
# Indice : if / else

# À toi :
if niveau_acces == 3:
    etat_porte = 'Ouverte'
else:
    etat_porte = 'Fermée'

# --- Vérification (ne pas modifier) ---
assert etat_porte == "Ouverte", f"Obtenu : '{etat_porte}'"
print("✅ Correct !")

File: 20260724/exos/02_condition.lua
-- ============================================================
-- The Last of Us — porte verrouillée
-- ============================================================
-- Si Joel a la clé, le message est "Ouverte". Sinon "Fermee".
--
-- Lance : lua 02_condition.lua
-- ============================================================

a_cle = true
etat = ""

-- Remplis etat selon a_cle.
--
-- a_cle = true   →  etat = "Ouverte"
-- a_cle = false  →  etat = "Fermee"
--
-- Résultat attendu : etat == "Ouverte"  (car a_cle vaut true ici)
--
-- Indice : if / else / end

-- À toi :
if a_cle == true then
    etat = "Ouverte"
else
    etat = "Fermee"
end

-- --- Vérification (ne pas modifier) ---
assert(etat == "Ouverte", "Obtenu : '" .. etat .. "'")
print("✅ Correct !")
